### ISSUE TITLE
Bug: [no-deprecated] doesn't report deprecated imported values used in object shorthand properties

### ISSUE THREAD

**Issue by ulrichstark** [typescript-eslint/typescript-eslint#12774]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.2&fileType=.tsx&code=HYQwtgpgzgDiDGEAEAbAlgIyQbwFBKQHoAqYpAAQBMIYAnCeEAFwkqWMPyQgA8YB7WkyTx%2BwKMP4o2AXiQBGANy4AvrlxowAoUimzUmAHR7luAG780bbLulIVigk%2BdFCSNMFG168JigCeSMD8wvTaLJTmlta2lABcsfaOzoRuYYIRQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
namespace lib {
  /** @deprecated */
  export const old = 1;
}

import old = lib.old;

void { old };          // incorrectly not reported
void { old: old };     // reported
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Report the `void { old }` line

### Actual Result

Doesn't report the `void { old }` line

### Additional Info

This was originally observed with a standard ES import from @tanstack/react-table:

```ts
import { sortFns } from "@tanstack/react-table";

const options = { sortFns }; // not reported
```

The namespace in the reproduction only makes the issue self-contained for the playground.

**Comment by StyleShit** [comment]
Ha, Interesting!

Thanks for reporting, and also for including the original snippet

BTW, the playground supports 3rd party packages 😄
[Playground](https://typescript-eslint.io/play/#ts=6.0.2&fileType=.tsx&code=JYWwDg9gTgLgBAbzgZ2jAYgO2XAvnAMyghDgCIABGAQ2xoGMBrAeigFNr6YBaGgIwA2bMgG4AUGPoQ6cCGBjBpOALyIUaLDlwi4zZnEwR47SLDYATIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by kirkwaiblinger** [comment]
Is this a different issue than aliasing in general? I vaguely recall there were some known unfixable issues in this area.

[Playground](https://typescript-eslint.io/play/#ts=6.0.2&fileType=.tsx&code=PQKhAIAEBMFMAcBOsDGBDALra4TAFAoD2AdgM4bhxKqbbgC84JArgDZsCEA3IaWUTawAdGyIBzABQBvKgmTosOAL4BKbuGDBwxRAoz4%2B5SmjYBLNGQDyAMwAi82ksZyai7L2LlBIsVNmmFtb2ju4q6praZAAWROw4yPBEiBhAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFADuQ6E0jgwAXxCKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

### Code

```ts
/** @deprecated */
const deprecated = null!;
console.log({ deprecated }); // correct

const aliasOfDeprecated = deprecated;
console.log({ aliasOfDeprecated }); // should report
```
